lib/rules/lint-require-valid-alt-text.js

```diff
@@ -0,0 +1,98 @@
+      column: node.loc && node.loc.start.column,
+      source: this.sourceForNode(node),
+    });
+  }
+  visitor() {
+    return {
+      ElementNode(node) {
+        const isImg = AstNodeInfo.isImgElement(node);
+        const isObj = AstNodeInfo.isObjectElement(node);
+        const isInput = AstNodeInfo.isInputElement(node);
+        const isArea = AstNodeInfo.isAreaElement(node);
+        const hasAriaHiddenAttribute = AstNodeInfo.hasAttributeValue(node, 'aria-hidden', 'true');
+        if (hasAriaHiddenAttribute) {
+          return;
+        }
+        if (isImg) {
+          const hasAltAttribute = AstNodeInfo.hasAttribute(node, 'alt');
+          const altValue = AstNodeInfo.elementAttributeValue(node, 'alt');
+          const roleValue = AstNodeInfo.elementAttributeValue(node, 'role');
+          if (!hasAltAttribute) {
+            this.logNode({
+              message:
+                'img tags must have an alt attribute, either with meaningful text, or an empty string for decorative images.',
+              node,
+            });
+          } else {
+            if (altValue === '') {
+              if (['presentation', 'none'].includes(roleValue)) {
+                return;
+              }
+              this.logNode({
+                message:
+                  'img tags must have an alt attribute, either with meaningful text, or an empty string for decorative images.',
+                node,
+              });
+            }
+          }
+        } else if (isInput) {
+          const isImageInput = AstNodeInfo.hasAttributeValue(node, 'type', 'image');
+          if (!isImageInput) {
+            return;
+          }
+          const hasValidAttributes = AstNodeInfo.hasAnyAttribute(node, [
+            'aria-label',
+            'aria-labelledby',
+            'alt',
+          ]);
+
+          if (!hasValidAttributes) {
+            this.logNode({
+              message:
+                '<input> elements with type="image" must have a text alternative through the `alt`, `aria-label`, or `aria-labelledby` attribute.',
+              node,
+            });
+          }
+        } else if (isObj) {
+          const hasValidAttributes = AstNodeInfo.hasAnyAttribute(node, [
+            'aria-label',
+            'aria-labelledby',
+            'title',
+          ]);
+
+          if (hasValidAttributes || hasAccessibleChild(node)) {
+            return;
+          }
+
+          this.logNode({
+            message:
+              'Embedded <object> elements must have alternative text by providing inner text, aria-label or aria-labelledby attributes.',
+            node,
+          });
+        } else if (isArea) {
+          if (!AstNodeInfo.hasAnyAttribute(node, ['aria-label', 'aria-labelledby', 'alt'])) {
+            this.logNode({
+              message:
+                'Each area of an image map must have a text alternative through the `alt`, `aria-label`, or `aria-labelledby` attribute.',
```

Comment:
please add "See https://www.w3.org/WAI/WCAG21/Techniques/html/H24 for more information."